v2/pkg/astvisitor/visitor.go

```diff
@@ -124,6 +143,32 @@ func WalkerFromPool() *Walker {
 	return walker.(*Walker)
 }
 
+func WalkerFromPool2(pool string) *Walker {
+	switch pool {
+	case "A":
+		walker := walkerPoolA.Get()
+		if walker == nil {
+			w := NewWalkerWithID(8, "RepresentationVariables")
+			return &w
+		}
+		return walker.(*Walker)
+	case "B":
+		walker := walkerPoolB.Get()
+		if walker == nil {
+			w := NewWalkerWithID(8, "CollectNodes")
+			return &w
+		}
+		return walker.(*Walker)
+	default:
+		walker := walkerPool.Get()
+		if walker == nil {
+			w := NewWalker(8)
+			return &w
+		}
+		return walker.(*Walker)
+	}
+}
+
```

Comment:
This function might be doing too much. The only thing that is actually used here is the `case "A"` branch. Also for the pool we can define how to create a new instance of the object via the `New` method:
```go
walkerPoolA = sync.Pool{
	New: func() any {
		return new(NewWalkerWithID(8, "RepresentationVariables"))
	},
}
```

So the getting of new items would be as simple as:

```go
walker := walkerPoolA.Get().(*walker.Walker)
```
This might give you ideas how to simplify the managment around pools.